test/ios/MetricsTests.m

```diff
@@ -0,0 +1,228 @@
+#import "MGLMapboxEvents.h"
+
+#import "KIFTestActor+MapboxGL.h"
+
+#import <KIF/KIF.h>
+#import <OCMock/OCMock.h>
+
+#import "MapboxGL.h"
+#import "OHHTTPStubs.h"
+
+@interface MGLMapboxEvents (Testing)
+
++ (instancetype)sharedManager;
+- (NSString *)appBundleId;
+- (NSString *)token;
+- (BOOL)isPaused;
+- (NSURLSession *)session;
+- (void)pushEvent:(NSString *)event withAttributes:(NSDictionary *)attributeDictionary;
+- (NSMutableArray *)eventQueue;
+- (void)postEvents:(NSArray *)events;
+- (NSTimer *)timer;
+- (NSUInteger)flushAt;
+- (NSTimeInterval)flushAfter;
+- (void)setFlushAfter:(NSTimeInterval)newFlushAfter;
+- (void)flush;
+- (void)startTimer;
+
+@end
+
+@interface MetricsTests : KIFTestCase
+
+@property (nonatomic) NSTimeInterval defaultFlushAfter;
+
+@end
+
+@implementation MetricsTests
+
+- (void)beforeAll {
+    [tester acknowledgeSystemAlert];
+    self.defaultFlushAfter = [[MGLMapboxEvents sharedManager] flushAfter];
```

Comment:
#1491 turns `-[MGLMapboxEvents flushAfter]` into a constant, [`MGLFlushInterval`](https://github.com/mapbox/mapbox-gl-native/blob/1ec5-metrics-opt-way-out/platform/ios/MGLMapboxEvents.m#L48) because it wasn’t being modified anywhere. I’m fine with exposing the constant to these tests, but given that modifying the flush interval will no longer be supported, it won’t be necessary to test that case.